src/graph.py

```diff
@@ -0,0 +1,200 @@
+            raise ValueError('Edge does not exist.')
+
+    def has_node(self, n):
+        """Find a specific node in the graph."""
+        return n in self.g
+
+    def neighbors(self, n):
+        """Return all the edges for n."""
+        try:
+            nbrs = []
+            for i in self.g[n]:
+                nbrs += [i[0]]
+            return nbrs
+        except KeyError:
+            raise KeyError('No node exists.')
+
+    def adjacent(self, n1, n2):
+        """Return true if an edge exists between n1 and n2."""
+        try:
+            list1 = []
+            list2 = []
+            for i in self.g[n1]:
+                list1 += [i[0]]
+            for i in self.g[n2]:
+                list2 += [i[0]]
+            if n1 in list2 or n2 in list1:
+                return True
+            return False
+        except KeyError:
+            raise KeyError('No node exists.')
+
+    def depth_first_traversal(self, start):
+        """Return a list of nodes based on depth first traversal."""
+        start = (start, 0)
+        travel = [start]
+        depth_list = []
+        depth = set()
+        while travel:
+            edge = travel.pop()[0]
+            if edge not in depth:
+                travel.extend(self.g[edge][::-1])
+                depth_list.append(edge)
+                depth.add(edge)
+        return depth_list
+
+    def breadth_first_traversal(self, start):
+        """Return a list of nodes based on breadth first traversal."""
+        start = (start, 0)
+        travel = [start]
+        depth = []
+        while travel:
+            edge = travel.pop(0)[0]
+            if edge not in depth:
+                travel.extend(self.g[edge])
+                depth.append(edge)
+        return depth
+
+    def warshall(self, start, finish):
+        """Floyd Warshall formula."""
+        if self._path(start, finish):
+            distance = {}
+            path = {}
+            graph = self.g
+            for node in graph:
+                distance[node] = {}
+                path[node] = {}
+                for visit in graph:
+                    distance[node][visit] = 1000
+                    path[node][visit] = -1
+                distance[node][node] = 0
+                for edge in range(len(graph[node])):
+                    key_edge = graph[node][edge][0]
+                    distance[node][key_edge] = graph[node][edge][1]
+                    path[node][key_edge] = node
+            for second_node in graph:
+                for node in graph:
+                    for visit in graph:
+                        new_distance = distance[node][second_node] + distance[second_node][visit]
+                        if new_distance < distance[node][visit]:
+                            distance[node][visit] = new_distance
+                            path[node][visit] = path[second_node][visit]
+            return self._return_path_warthal(path, start, finish)
+        raise KeyError('No such path.')
```

Comment:
Why do you return a different type of error in each of your shortest path algorithms?